app/packs/review/_components/UnreviewedPack.tsx

```diff
@@ -8,41 +18,199 @@ import {
+import { Badge } from '@/components/ui/badge'
+import { packMap } from '@/types'
+import { useState } from 'react'
+
+import { PackType } from '@/types'
 
-export interface QuestionPackProps {
-	id: number
-	featured?: boolean // popular
+interface UnreviewedPackProps {
+	id: string
 	name: string
 	description: string
-	likes: number
 	type: string
-	questions: string[]
-	slug: string
+	questions: any[]
+	isDenied?: boolean
+	onStatusChange?: () => void
 }
 
-export default function UnreviewedPack({ pack }: { pack: QuestionPackProps }) {
+export default function UnreviewedPack({
+	id,
+	name,
+	description,
+	type,
+	questions,
+	isDenied,
+	onStatusChange
+}: UnreviewedPackProps) {
+	const [rejectionReason, setRejectionReason] = useState('')
+	const [isLoading, setIsLoading] = useState(false)
+	const [dialogState, setDialogState] = useState<'accept' | 'reject' | null>(
+		null
+	)
+
+	const handleAction = async (action: 'accept' | 'reject') => {
+		setIsLoading(true)
+		try {
+			const response = await fetch('/api/packs/review', {
+				method: 'PUT',
+				headers: {
+					'Content-Type': 'application/json'
+				},
+				body: JSON.stringify({
+					id,
+					approved: action === 'accept' ? true : false,
+					message: action === 'reject' ? rejectionReason : undefined
+				})
+			})
+
+			if (!response.ok) throw new Error('Failed to process request')
+
+			toast({
+				title: action === 'accept' ? 'Pack Accepted' : 'Pack Rejected',
+				description:
+					action === 'accept' ?
+						'The pack has been approved and is now public'
+					:	'The pack has been rejected and the author will be notified'
+			})
+
+			setDialogState(null)
+			setRejectionReason('')
+			onStatusChange?.()
+		} catch (error) {
+			toast({
+				title: 'Error',
+				description: 'Failed to process the request',
+				variant: 'destructive'
+			})
+		} finally {
+			setIsLoading(false)
+		}
+	}
+
+	const handleDialogClose = () => {
+		// Reset rejection reason when dialog closes
+		setRejectionReason('')
+		setDialogState(null)
+	}
+
 	return (
-		<li className="p-[3px]">
-			<Card className="border-none h-full">
-				<CardHeader className="relative">
-					<CardTitle>{pack.name}</CardTitle>
-					<CardDescription>{pack.description}</CardDescription>
-				</CardHeader>
+		<Card className="relative border-none h-full flex flex-col justify-between">
+			<CardHeader>
+				<CardTitle>{name}</CardTitle>
+				<CardDescription>{description}</CardDescription>
+			</CardHeader>
+			<div>
 				<CardContent className="grid grid-cols-2">
 					<div className="flex flex-col gap-0.5">
 						<h3 className="text-sm text-muted-foreground">Questions</h3>
-						<p className="">{pack.questions.length}</p>
+						<p className="">{questions.length}</p>
 					</div>
 					<div className="flex flex-col gap-0.5">
 						<h3 className="text-sm text-muted-foreground">Type</h3>
-						<p className="">{pack.type}</p>
+						<p className="">{packMap[type as PackType]}</p>
```

Comment:
This line we can get rid of the extra casting with the above type changes.

Also, nit: we can get rid of the `className=""` since its empty
```suggestion
						<p >{packMap[type]}</p>
```